cmd/influxd/server_integration_test.go

```diff
@@ -605,3 +628,63 @@ func Test_ServerMultiLargeBatchIntegration(t *testing.T) {
 	}
 	simpleCountQuery(t, testName, nodes, `select count(value) from "foo"."bar".cpu`, "value", batchSize*int64(nBatches))
 }
+
+func Test_ServerSingleGraphiteIntegration(t *testing.T) {
+	//t.Skip()
+	nNodes := 1
+	basePort := 8790
+	testName := "graphite integration"
+	now := time.Now().UTC().Round(time.Millisecond)
+	c := main.NewConfig()
+	g := main.Graphite{
+		Enabled:  true,
+		Database: "graphite",
+		Protocol: "TCP",
+	}
+	c.Graphites = append(c.Graphites, g)
+
+	t.Logf("Graphite Connection String: %s\n", g.ConnectionString(c.BindAddress))
+	nodes := createCombinedNodeCluster(t, testName, nNodes, basePort, c)
+
+	createDatabase(t, testName, nodes, "graphite")
+	createRetentionPolicy(t, testName, nodes, "graphite", "raw")
+
+	// Connect to the graphite endpoint we just spun up
+	conn, err := net.Dial("tcp", g.ConnectionString(c.BindAddress))
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	// Get the current index
+	i := index(t, testName, nodes)
+
+	t.Log("Writing data")
+	data := []byte(`cpu 12.345 `)
+	data = append(data, []byte(fmt.Sprintf("%d", now.UnixNano()/1000000))...)
+	data = append(data, '\n')
+	_, err = conn.Write(data)
+	conn.Close()
+	if err != nil {
+		t.Fatal(err)
+		return
+	}
+
+	// Wait for sync
+	wait(t, testName, nodes, i+4)
```

Comment:
I actually didn't track it down to find out.  I kept incrementing it until it failed :-).  The real problem calculating this is that writing data doesn't correlate to a single index.  And in fact, this could break if we change underlying code.  What I really would like is something that tells me this data is committed, not that I'm at a specific index (which has little meaning to the action I requested).  Do you have thoughts on this?